From db8c33840b06f8cb404dac641ebd604b5662878b Mon Sep 17 00:00:00 2001
From: Don Darling <don.osc2@gmail.com>
Date: Thu, 11 Mar 2010 13:24:26 -0600
Subject: [PATCH] Disable support for atomic spinlocks.

Support for arm atomic spinlocks appeared in 2.14.1 but consistently crashes
DM6446 boards.  Disable until problem can be resolved.  There is currently no
way to disable this with a configure option so we must patch the config.h file.

Converted from glib1_2_22_2.patch
---
 config.h     |   10 ++++++++--
 glibconfig.h |    2 ++
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/config.h b/config.h
index 510b7f2..416c48c 100644
--- a/config.h
+++ b/config.h
@@ -64,7 +64,13 @@
 /* #undef G_ATOMIC_ALPHA */
 
 /* arm atomic implementation */
-#define G_ATOMIC_ARM 1
+/* #undef G_ATOMIC_ARM */
+
+ /* DJD: support for arm atomic spinlocks appeared in 2.14.1 but consistently 
+  * crashes DM6446 boards.  Disable until problem can be resolved.  There is
+  * currently no way to disable this with a configure option so we must patch
+  * the config.h file.
+  */
 
 /* cris atomic implementation */
 /* #undef G_ATOMIC_CRIS */
@@ -362,7 +368,7 @@
 /* #undef HAVE_SELINUX */
 
 /* Define to 1 if you have the <selinux/selinux.h> header file. */
-/* #undef HAVE_SELINUX_SELINUX_H */
+#define HAVE_SELINUX_SELINUX_H 1
 
 /* Define to 1 if you have the `setenv' function. */
 #define HAVE_SETENV 1
diff --git a/glibconfig.h b/glibconfig.h
index e8db5a7..869346d 100644
--- a/glibconfig.h
+++ b/glibconfig.h
@@ -167,6 +167,8 @@ union _GSystemThread
   long   dummy_long;
 };
 
+#define G_ATOMIC_OP_MEMORY_BARRIER_NEEDED 1
+
 #define GINT16_TO_LE(val)	((gint16) (val))
 #define GUINT16_TO_LE(val)	((guint16) (val))
 #define GINT16_TO_BE(val)	((gint16) GUINT16_SWAP_LE_BE (val))
-- 
1.6.3.3

